### DISCUSSION TITLE
question on `brew services` output

### DISCUSSION THREAD

**Discussion by rkazak** [Homebrew/discussions#2858] [Everyday usage]
When I run the following command I get an 'error' as the status,

```
rkazak@mbp ~ % brew services -v    
Name    Status     User File
dnsmasq error  512 root ~/Library/LaunchAgents/homebrew.mxcl.dnsmasq.plist
unbound none
```

However when I run it via `sudo` I get the status I expect,

```
rkazak@mbp ~ % sudo brew services -v
Password:
Name    Status  User File
dnsmasq started root /Library/LaunchDaemons/homebrew.mxcl.dnsmasq.plist
unbound none         
```

Is this working as expected or a bug?

**Comment by SMillerDev** [comment]
Well, those are two different results. Clearly it failed to run as user but succeeded as root.

**Reply by rkazak** [reply]
That is true, however 'error' is misleading, better to have a msg requesting user to run as root for root based services and not show it at all.

**Reply by carlocab** [reply]
That might be better indeed. Pull requests welcome: https://github.com/Homebrew/homebrew-services

**Reply by SMillerDev** [reply]
That message is already there, though it could probably be improved. The only thing is that it's not blocked so you can still run the service as a user.
```console
~ $ brew services start dnsmasq
Warning: dnsmasq must be run as root to start at system startup!
==> Successfully started `dnsmasq` (label: homebrew.mxcl.dnsmasq)
```

**Comment by peimn** [comment]
This happened to me with DNSmasq as well, and it is possible you may receive this `error 512` if you include an IP address that isn't reachable in the `listen-address` config.
